openedx/openedx-platform
# feat: add new library util for split studio view

## Description

This PR adds an catch for libraries when `split_library_view_on_dashboard` is enabled. Currently in the `course-authoring` MFE on stage and prod, no libraries populate because `split_library_view_on_dashboard` is enabled. This PR adds the libraries to the JSON response when the request format is  `json` and `split_library_view_on_dashboard` is enabled. The legacy behavior of keeping the library data fetch separate when `split_library_view_on_dashboard` is enabled is preserved. This change impacts Course Authors.

## Supporting information

JIRA Ticket: [TNL-11091](https://2u-internal.atlassian.net/browse/TNL-11091)

## Testing instructions

1. Navigate to the [Studio home MFE page](http://localhost:2001/home)
2. Click on the libraries tab
3. Check that expected libraries load
4. Navigate to [Django admin](http://localhost:18010/admin/) waffle flags and enable `contentstore.split_library_on_studio_dashboard`
5. Navigate to  back the [Studio home MFE page](http://localhost:2001/home)
6. Click on the libraries tab
7. Check that the expected libraries load

Repeat the steps above for the [legacy Studio home page](http://localhost:18010/home/)

## Deadline

None
